modules/products/confluence/variables.tf

```diff
@@ -99,4 +99,32 @@ variable "pvc_claim_name" {
 variable "enable_synchrony" {
   description = "If true, Collaborative editing service will be enabled."
   type        = bool
+}
+
+variable "db_snapshot_identifier" {
+  description = "Snapshot identifier for RDS."
+  type        = string
+  default     = null
+}
+
+variable "db_master_username" {
+  description = "Master username for the RDS instance."
+  type        = string
+  default     = null
+}
+
+variable "db_master_password" {
+  description = "Master password for the RDS instance."
+  type        = string
+  default     = null
+}
+
+variable "db_snapshot_build_number" {
+  description = "Confluence build number of the database snapshot."
+  type        = string
+  default     = null
+  validation {
+    condition     = var.db_snapshot_build_number == null || can(regex("^[0-9]{4}$", var.db_snapshot_build_number))
+    error_message = "Build number must be provided in the right format when using snapshot to restore database."
```

Comment:
If the build number is exact 4 digit, then we need to mention in validation error message. This message is not clear enough.